python-package/lightgbm/basic.py

```diff
@@ -633,7 +636,8 @@ def _lazy_init(self, data, label=None, max_bin=255, reference=None,
         params = {} if params is None else params
         self.max_bin = max_bin
         self.predictor = predictor
-        params["max_bin"] = max_bin
+        if self.max_bin is not None:
```

Comment:
@wxchan Please add deprecation warning here.

>StrikerRUS 23 hours ago Collaborator 
>Maybe than it's better to completely remove max_been from arguments (for 1 or 2 release mark as deprecated)? Because it's oddly that None actually means 255...

>guolinke an hour ago Member 
>I think we can remove them from arguments as well